packages/aws-cdk-lib/aws-codebuild/lib/fleet.ts

```diff
@@ -0,0 +1,164 @@
+  /**
+   * Creates a Fleet construct that represents an external fleet.
+   *
+   * @param scope The scope creating construct (usually `this`).
+   * @param id The construct's id.
+   * @param fleetArn arn of external fleet.
+   */
+  public static fromFleetArn(scope: Construct, id: string, fleetArn: string): IFleet {
+    class Import extends Resource implements IFleet {
+      public readonly fleetName = Arn.split(fleetArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!.split(':')[0];
+      public readonly fleetArn = fleetArn;
+
+      public get computeType(): FleetComputeType {
+        throw new Error('Cannot retrieve computeType property from an imported Fleet');
+      }
+      public get environmentType(): EnvironmentType {
+        throw new Error('Cannot retrieve environmentType property from an imported Fleet');
+      }
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * The ARN of the fleet
+   */
+  public readonly fleetArn: string;
+
+  /**
+   * The name of the fleet
+   */
+  public readonly fleetName: string;
+
+  /**
+   * The compute type of the fleet.
+   */
+  public readonly computeType: FleetComputeType;
+
+  /**
+   * The environment type of the fleet.
+   */
+  public readonly environmentType: EnvironmentType;
+
+  constructor(scope: Construct, id: string, props: FleetProps) {
+    if (props.fleetName && !Token.isUnresolved(props.fleetName)) {
+      if (props.fleetName.length < 2) {
+        throw new Error(`Fleet name can not be shorter than 2 characters but has ${props.fleetName.length} characters.`);
+      }
+      if (props.fleetName.length > 128) {
+        throw new Error(`Fleet name can not be longer than 128 characters but has ${props.fleetName.length} characters.`);
+      }
+    }
+
+    super(scope, id, { physicalName: props.fleetName });
+
+    if ((props.baseCapacity ?? 1) < 1) {
+      throw new Error('baseCapacity must be greater than or equal to 1');
+    }
+
+    const resource = new CfnFleet(this, 'Resource', {
+      name: props.fleetName,
+      baseCapacity: props.baseCapacity,
+      computeType: props.computeType,
+      environmentType: props.environmentType,
+    });
+
+    this.fleetArn = this.getResourceArnAttribute(resource.attrArn, {
+      service: 'codebuild',
+      resource: 'fleet',
+      resourceName: this.physicalName,
+      arnFormat: ArnFormat.SLASH_RESOURCE_NAME,
+    });
+    this.fleetName = this.getResourceNameAttribute(resource.ref);
+    this.computeType = props.computeType;
+    this.environmentType = props.environmentType;
+  }
+}
+
+/**
+ * Fleet build machine compute type. Subset of Fleet compatible {@link ComputeType} values.
+ *
+ * @see https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-compute-types.html#environment.types
+ */
+export enum FleetComputeType {
+  SMALL = ComputeType.SMALL,
+  MEDIUM = ComputeType.MEDIUM,
+  LARGE = ComputeType.LARGE,
+  X_LARGE = ComputeType.X_LARGE,
+  X2_LARGE = ComputeType.X2_LARGE,
+}
```

Comment:
see note above. worth moving all the enums into `enums.ts` ?

Reply:
See https://github.com/aws/aws-cdk/pull/29754#discussion_r1564123370